transcendence-app/src/chat/chat.gateway.ts

```diff
@@ -9,75 +11,72 @@ import {
   SubscribeMessage,
   WebSocketGateway,
   WebSocketServer,
+  WsException,
 } from '@nestjs/websockets';
 import { Socket, Server } from 'socket.io';
-import { v4 as uuidv4 } from 'uuid';
+import { BadRequestTransformationFilter } from '../shared/filters/bad-request-transformation.filter';
 import { WsAuthenticatedGuard } from '../shared/guards/ws-authenticated.guard';
-import { User } from '../user/infrastructure/db/user.entity';
-
-type ChatroomMessage = {
-  id: string;
-  user: User;
-  content: string;
-  createdAt: number;
-  chatroomId: string;
-};
-
-type ChatroomId = string;
+import { ChatService } from './chat.service';
+import { ChatroomMemberService } from './chatroom/chatroom-member/chatroom-member.service';
+import { CreateChatroomMessageDto } from './chatroom/chatroom-message/dto/create-chatroom-message.dto';
 
 @WebSocketGateway({ path: '/api/v1/socket.io' })
 @UseGuards(WsAuthenticatedGuard)
 @UseInterceptors(ClassSerializerInterceptor)
+@UseFilters(BadRequestTransformationFilter)
 export class ChatGateway {
   @WebSocketServer() server!: Server;
-  chatrooms = new Map<ChatroomId, Set<ChatroomMessage>>();
+
+  constructor(
+    private chatService: ChatService,
+    private chatroomMemberService: ChatroomMemberService,
+  ) {}
 
   @SubscribeMessage('chatroomMessage')
-  handleMessage(
-    @MessageBody() data: { chatroomId: ChatroomId; content: string },
+  async handleMessage(
+    @MessageBody()
+    createChatroomMessageDto: CreateChatroomMessageDto,
     @ConnectedSocket() client: Socket,
   ) {
+    const { chatroomId, content } = createChatroomMessageDto;
     const user = client.request.user;
-    const { chatroomId, content } = data;
-    const chatroom = this.chatrooms.get(chatroomId);
-    if (user && chatroom) {
-      const message: ChatroomMessage = {
-        id: uuidv4(),
-        user: user,
-        content: content,
-        createdAt: Date.now(),
-        chatroomId: chatroomId,
-      };
-      chatroom.add(message);
-      this.server.to(chatroomId).emit('chatroomMessage', message);
+    const chatroomMember = await this.chatroomMemberService.getById(
+      chatroomId,
+      user.id,
+    );
+    if (!chatroomMember || chatroomMember.muted) {
+      throw new WsException('Forbidden');
```

Comment:
We have to remember to n ot silently fail. We have to somehow prevent this situation from normally ever happening when navigating the site.
From the scale:
> [...] there must be no unhandled warning or error

Reply:
Updated. Thanks for the suggestion.